pybind_interface/pybind_main.cpp

```diff
@@ -489,6 +556,132 @@ py::array_t<float> qsim_simulate_fullstate(
   return py::array_t<float>(fsv_size, fsv, capsule);
 }
 
+py::array_t<float> qtrajectory_simulate_fullstate(const py::dict &options,
+                                                  uint64_t input_state) {
+  NoisyCircuit<Cirq::GateCirq<float>> ncircuit;
+  unsigned num_qubits;
+  try {
+    ncircuit = getNoisyCircuit(options);
+    num_qubits = qsim::count_qubits(ncircuit);
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  using Simulator = qsim::Simulator<For>;
+  using StateSpace = Simulator::StateSpace;
+  using State = StateSpace::State;
+  using Runner = qsim::QuantumTrajectorySimulator<IO, Cirq::GateCirq<float>,
+                                                  MultiQubitGateFuser,
+                                                  Simulator>;
+
+  Runner::Parameter param;
+  uint64_t seed;
+  try {
+    param.num_threads = parseOptions<unsigned>(options, "t\0");
+    param.max_fused_size = parseOptions<unsigned>(options, "f\0");
+    param.verbosity = parseOptions<unsigned>(options, "v\0");
+    seed = parseOptions<unsigned>(options, "s\0");
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  StateSpace state_space(param.num_threads);
+
+  float *fsv;
+  const uint64_t fsv_size = std::pow(2, num_qubits + 1);
+  const uint64_t buff_size = state_space.MinSize(num_qubits);
+  if (posix_memalign((void **)&fsv, 32, buff_size * sizeof(float))) {
+    IO::errorf("Memory allocation failed.\n");
+    return {};
+  }
+
+  State state = state_space.Create(fsv, num_qubits);
```

Comment:
Related to above question: why use this creation method instead of the default  Create() ?

Reply:
This is copied from the analogous `qsim_simulate_fullstate` method. + @sergeisakov for insight on copied code.